app/src/main/java/me/hufman/androidautoidrive/music/controllers/SpotifyAppController.kt

```diff
@@ -292,16 +384,22 @@ class SpotifyAppController(context: Context, val remote: SpotifyAppRemote, val w
 	}
 
 	/**
-	 * Creates a new instance of the [QueueMetadata] with the queue cover art loaded.
+	 * Updates the [QueueMetadata] with the cover art of the current queue. If the
+	 * [TemporaryPlaylistState] and its [AppSettings.KEYS] key are supplied then it will be updated
+	 * with the result.
 	 */
-	private fun loadQueueCoverart() {
+	private fun loadQueueCoverart(temporaryPlaylistState: TemporaryPlaylistState? = null, appSettingsTemporaryPlaylistStateKey: AppSettings.KEYS? = null) {
```

Comment:
I don't like this code smell, which signals that this function is doing too much work: I believe the extra complication is to save the current queue's coverArtUri to the TemporaryPlaylistState. I wonder if this extra retention is actually necessary, however.
When the user plays the Liked Songs playlist, this loadQueueCoverart will load the proper heart banner normally. After the user clicks a queue entry and the context switches to the temporary playlist, this function will return the normal Spotify coverart for the temporary playlist, which might be fine in my estimation. Alternatively, the recentlyPlayed results could skip the TemporaryPlaylist if it's the first result and assume that the next item is the real Liked Songs.
However if the TemporaryPlaylistState really wants to hold its source coverArtUri, I'd prefer to have `createTemporaryPlaylistState` always do the regular `loadQueueCoverart` first (additionally allowing Car UI to show the queue cover art during any playlist synchronization) and then create and cache the entire coherent object using the real queueCoverArt information, instead of trying to recover a missing coverArtUri later in `loadTemporaryPlaylistState`. Such guessing will be even harder when Artist queues get added. The calculated queueCoverArtUri can go live as an additional member at the top with the queueTitle for the wrapping `createTemporaryPlaylistState` to use.